### DISCUSSION TITLE
dependent boundary for hyperparameter space in optuna.study when using trial.suggest

### DISCUSSION THREAD

**Discussion by AMagenschab** [optuna/optuna#4092] [Q&A] [answered]
Hi all, 
is it actually possible to set a dependent boundary for the hyperparameterspace when using trial.suggest in an optuna.study?

E.g. num_leaves should be smaller or equal the 2^(the selected max_depth of the trial)
trying: 
params = {
    "space": trial.suggest_int('num_leaves': (20, 2^(trial.suggest_int('max_depth': (3,30))
}

does not work out.
Is there another workaround you can recommen? I did not find any relevant info on that.

In the hyperopt package  that seems to be possible:
https://stackoverflow.com/questions/53041467/hyperopt-define-parameter-which-is-dependent-on-other-parameter

Thanks!
BR,

**Comment by not522** [comment, marked as answer]
I think what you want to do is the following.

```python
max_depth = trial.suggest_int('max_depth', 3, 30)
num_leaves = trial.suggest_int('num_leaves', 20, 2**max_depth)
params = {
    "space": {
        "max_depth": max_depth,
        "num_leaves": num_leaves,
    }
}
```

**Reply by AMagenschab** [reply]
Thank you very much!
works perfectly fine!
